src/sources/opentelemetry/tests.rs

```diff
@@ -1106,120 +1107,250 @@ async fn receive_summary_metric() {
-            .header("User-Agent", "Test")
-            .body(req.encode_to_vec())
+
+        let res = client
+            .post(format!("http://{}/v3/logs", env.http_addr))
+            .header::<&str, String>("Content-Type", "application/json".into())
+            .json(&payload)
             .send()
             .await
             .expect("Failed to send log to Opentelemetry Collector.");
 
-        let mut output = test_util::collect_ready(logs_output).await;
-        assert_eq!(output.len(), 1);
-        let actual_event = output.pop().unwrap();
-        schema_definitions
-            .unwrap()
-            .assert_valid_for_event(&actual_event);
-        let expect_vec = vec_into_btmap(vec![
-            ("AbsentHeader", Value::Null),
-            ("User-Agent", "Test".into()),
-            ("message", "log body".into()),
-            ("trace_id", "4ac52aadf321c2e531db005df08792f5".into()),
-            ("span_id", "0b9e4bda2a55530d".into()),
-            ("severity_number", 9.into()),
-            ("severity_text", "info".into()),
-            ("flags", 4.into()),
-            ("dropped_attributes_count", 0.into()),
-            ("timestamp", Utc.timestamp_nanos(1).into()),
-            ("observed_timestamp", Utc.timestamp_nanos(2).into()),
-            ("source_type", "opentelemetry".into()),
-        ]);
-        let mut expect_event = Event::from(LogEvent::from(expect_vec));
-        expect_event.set_upstream_id(Arc::new(OutputId {
-            component: "test".into(),
-            port: Some("logs".into()),
-        }));
-        assert_eq!(actual_event, expect_event);
+        assert_eq!(res.status(), hyper::StatusCode::NOT_FOUND);
+        assert_eq!(res.text().await.unwrap(), "Not found");
+    })
+    .await;
+}
+
+#[tokio::test]
+async fn delivery_errored() {
+    /*
+    Verify that on `BatchStatus::Errorred` we get a retryable status code
+    and a useful status message
+
+    Spec:
+
+    If the processing of the request fails, the server MUST respond with
+    appropriate HTTP 4xx or HTTP 5xx status code. See the sections below for
+    more details about specific failure cases and HTTP status codes that should
+    be used.
+
+    The response body for all HTTP 4xx and HTTP 5xx responses MUST be a
+    Protobuf-encoded Status message that describes the problem.
+
+    */
+    assert_source_compliance(&SOURCE_TAGS, async {
+        let env = build_otlp_test_env(LOGS, None, EventStatus::Errored).await;
+
+        let client = reqwest::Client::new();
+        let payload = ExportLogsServiceRequest {
+            resource_logs: vec![create_resource_logs("test")],
+        };
+
+        let post = client
+            .post(format!("http://{}/v1/logs", env.http_addr))
+            .header::<&str, String>("Content-Type", "application/json".into())
+            .json(&payload)
+            .send();
+
+        let (res, _out) = tokio::join!(post, test_util::collect_ready(env.output));
```

Comment:
I was hoping this would be enough to end up in the `BatchStatus::Errored` flow, but it seems to end up in the `StreamClosedError`

Any tips on how I can test the latter?